pandas/core/frame.py

```diff
@@ -3731,6 +3736,11 @@ def reindexer(value):
 
             # upcast
             value = cast_scalar_to_array(len(self.index), value)
+
+            # if extension dtype, value will be a list of length 1
```

Comment:
what is this?

Reply:
In this case, we simply want a 1d array, so again our two cases are:

1. `data` is a non-extension type, in which case `cast_scalar_array` will return a numpy array of width 1, which is what we want, in this case we don't need to do anything
2. `data` is an extension type, in which case `cast_scalar_array` will return a list containing a single extension array. In this case, we just want the extension array, so we index into the list and grab the only element.